code/game/gamemodes/objectives/_objective.dm

```diff
@@ -249,19 +262,22 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 					if (!(area.area_flags & HIDDEN_STASH_LOCATION))
 						continue
 					//This airlock is good for us
-					A.AddComponent(/datum/component/stash, receiver, secret_bag)
+					var/stash_comp = A.AddComponent(/datum/component/stash, owners, secret_bag)
+					for (var/datum/mind/receiver in owners)
+						if (!islist(receiver.antag_stashes))
+							receiver.antag_stashes = list()
+						receiver.antag_stashes += stash_comp
 					break
 		//Failsafe
 		if(!secret_bag.loc)
 			atom_text = "You"
-			message_admins("Could not find a location to put [ADMIN_FLW(receiver.current)]'s stash.")
-			secret_bag.forceMove(get_turf(receiver.current))
-			receiver.current.equip_to_appropriate_slot(secret_bag)
-			// Remove this from memory since the component deals with hard-dels
-			receiver.antag_stash = null
+			message_admins("Could not find a location to put [ADMIN_FLW(tester.current)]'s stash.")
+			secret_bag.forceMove(get_turf(tester.current))
+			tester.current.equip_to_appropriate_slot(secret_bag)
 		//Update the mind
-		receiver.store_memory("You have a secret stash of items hidden on the station required for your objectives. It is hidden inside of [atom_text] ([secret_bag.loc]) located at [get_area(secret_bag.loc)] [COORD(secret_bag.loc)], you may have to search around for it. (Use alt click on the object the stash is inside to access it).")
-		to_chat(receiver?.current, "<span class='notice bold'>You have a secret stash at [get_area(secret_bag)], more details are stored in your notes. (IC > Notes)</span>")
+		for (var/datum/mind/receiver in owners)
+			receiver.store_memory("You have a secret stash of items hidden on the station required for your objectives. It is hidden inside of [atom_text] ([secret_bag.loc]) located at [get_area(secret_bag.loc)] [COORD(secret_bag.loc)], you may have to search around for it. (Use alt click on the object the stash is inside to access it).")
+			to_chat(receiver?.current, "<span class='notice bold'>You have a secret stash at [get_area(secret_bag)], more details are stored in your notes. (IC > Notes)</span>")
```

Comment:
I think there should be signalRegister with MobLogin for those who are somehow DC'ed at the moment